psydac/linalg/topetsc.py

```diff
@@ -32,22 +32,27 @@ def flatten_vec( vec ):
         data = vec._data[idx].flatten()
         vec = coo_matrix(
                     (data,(indices,indices)),
-                    shape = [np.prod(npts),np.prod(npts)],
+                    shape = [vec.space.dimension,vec.space.dimension],
                     dtype = vec.space.dtype)
 
     elif isinstance(vec, BlockVector):
-        vecs = [vec_tocoo(b) for b in vec.blocks]
+        vecs = [flatten_vec(b) for b in vec.blocks]
+        vecs = [coo_matrix((v[1],(v[0],v[0])),
+                shape=[vs.space.dimension,vs.space.dimension],
+                dtype=vs.space.dtype) for v,vs in zip(vecs, vec.blocks)]
+
         blocks = [[None]*len(vecs) for v in vecs]
         for i,v in enumerate(vecs):
             blocks[i][i] = v
+
         vec = bmat(blocks,format='coo')
 
     else:
         raise TypeError("Expected StencilVector or BlockVector, found instead {}".format(type(vec)))
 
     array   = vec.data
     indices = vec.row
-    return array, indices
+    return indices, array
```

Comment:
If you swap the order of the two return arguments, you should do the same in the docstring

Reply:
fixed in [3e5c7df](https://github.com/pyccel/psydac/pull/244/commits/3e5c7df367d342592a91c6af5205579c80b5bbfb)